python/tests/test_batch.py

```diff
@@ -1368,3 +1454,200 @@ async def test_transaction_function_dump_restore(
+    async def test_batch_raise_on_error(
+        self, glide_client: GlideClusterClient, is_atomic: bool
+    ):
+        key = get_random_string(10)
+        # Ensure key2 is in the same slot as key for atomic transactions
+        key2 = f"{{{key}}}:{get_random_string(10)}"  # type: ignore # noqa
+
+        batch = (
+            ClusterBatch(is_atomic=is_atomic)
+            if isinstance(glide_client, GlideClusterClient)
+            else Batch(is_atomic=is_atomic)
+        )
+
+        batch.set(key, "hello")
+        batch.lpop(key)
+        batch.delete([key])
+        batch.rename(key, key2)
+
+        # Test with raise_on_error=False
+        result = await exec_batch(glide_client, batch, raise_on_error=False)
+
+        assert result is not None
+        assert len(result) == 4
+
+        assert result[0] == "OK"  # set(key, "hello")
+
+        assert isinstance(result[1], RequestError)  # lpop(key)
+        assert "WRONGTYPE" in str(result[1])
+
+        assert result[2] == 1  # delete([key])
+
+        assert isinstance(result[3], RequestError)  # rename(key, key2)
+        assert "no such key" in str(result[3])
+
+        # Test with raise_on_error=True
+        with pytest.raises(RequestError) as e:
+            await exec_batch(glide_client, batch, raise_on_error=True)
+
+        assert "WRONGTYPE" in str(e.value)
+
+    @pytest.mark.parametrize("cluster_mode", [True])
+    @pytest.mark.parametrize("is_atomic", [True, False])
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_cluster_batch_route(
+        self,
+        glide_client: GlideClusterClient,
+        protocol: ProtocolVersion,
+        is_atomic: bool,
+    ):
+        if await check_if_server_version_lt(glide_client, "6.0.0"):
+            pytest.skip("CONFIG RESETSTAT requires redis >= 6.0")
+
+        assert await glide_client.config_resetstat() == OK
+
+        key = get_random_string(10)
+        value = "value"
+        value_bytes = value.encode()
+
+        batch = ClusterBatch(is_atomic=is_atomic)
+        batch.set(key, value)
+        batch.get(key)
+
+        route = SlotKeyRoute(slot_type=SlotType.PRIMARY, slot_key=key)
+        results = await glide_client.exec(
+            batch, raise_on_error=True, route=route, timeout=2000
+        )
+
+        assert results == [OK, value_bytes]
+
+        # Check that no MOVED error occurred by inspecting errorstats on all nodes
+        error_stats_dict = await glide_client.info(
+            sections=[InfoSection.ERROR_STATS], route=AllNodes()
+        )
+        assert isinstance(error_stats_dict, dict)
+
+        for node_address, node_info in error_stats_dict.items():
+            assert isinstance(node_info, bytes)
+            # Ensure the errorstats section indicates no errors reported for this test
+            # It should only contain the header line.
+            assert (
+                node_info.strip() == b"# Errorstats"
```

Comment:
```suggestion
                node_info.strip() == b""# Errorstats\r\n""
```

? idk you know better